slang-gfx.h

```diff
@@ -545,12 +545,13 @@ class IShaderObject : public ISlangUnknown
     virtual SLANG_NO_THROW Result SLANG_MCALL setCombinedTextureSampler(
         ShaderOffset const& offset, IResourceView* textureView, ISamplerState* sampler) = 0;
 
-        /// Manually setting the specialization arguments for the shader object, overriding
-        /// the default arguments computed from the sub-objects.
+        /// Manually overrides the specialization argument for the sub-object binding at `offset`.
         /// Specialization arguments are passed to the shader compiler to specialize the type
         /// of interface-typed shader parameters.
-    virtual SLANG_NO_THROW Result SLANG_MCALL
-        setSpecializationArgs(const slang::SpecializationArg* args, uint32_t count) = 0;
+    virtual SLANG_NO_THROW Result SLANG_MCALL setSpecializationArgs(
+        ShaderOffset const& offset,
+        const slang::SpecializationArg* args,
+        uint32_t count) = 0;
```

Comment:
I assume the `count` is required because of the structured-buffer case, where the structure type could have multiple interface-type fields?

Having to handle that case here is a little unfortunate when it won't come up in the 99% case, but probably not the end of the world.

Reply:
I was thinking about the case where you set an object to a `ParameterBlock<X>` binding range, where X is a struct that contains more than one interface typed fields, then there could be more than one arguments. You can argue that the user should call `setSpecializationArgs` on the parameter block object's interface-typed binding range instead, but this interface is not doing harm to allow the user to set things directly at higher level.